### PR TITLE
New metric: Perceptual Path Length

### LINKED ISSUE #1933: Add Perceptual Path Length
## 🚀 Feature

Add Perceptual Path Length (PPL) as described in this paper by NVIDIA:

[A Style-Based Generator Architecture for Generative Adversarial Networks](https://arxiv.org/pdf/1812.04948.pdf)

### Motivation

Add additional metrics to support the evaluation of generative models.

### Pitch

Authors from Snap, ETH Zurich, and Carnegie Mellon used PyTorch Lightning in the following paper, but opted to use [a metrics library](https://github.com/toshas/torch-fidelity/tree/master) other than torchmetrics. The only metric the authors needed that does not exist in torchmetrics is PPL.

[Autodecoding Latent 3D Diffusion Models](https://arxiv.org/pdf/2307.05445.pdf)

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #1933

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
:books: Documentation preview :books:: https://torchmetrics--1939.org.readthedocs.build/en/1939/

### CODE REVIEW

File: requirements/image.txt [hunk 1]
```diff
@@ -3,5 +3,5 @@
 
 scipy >1.0.0, <1.11.0
 torchvision >=0.8, <=0.15.2
-torch-fidelity <=0.3.0
+torch-fidelity @ git+https://github.com/toshas/torch-fidelity@master
```
Comment (Borda) [inline requirements/image.txt:6]: note that this req. will not be listed in package dependency if installed from PyPI

**Review by Borda — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: requirements/image.txt (hunk 1 above)
Comment (Borda) [reply requirements/image.txt:6]: Yes, but what do you have in mind, even prevent PR to be merged with this dependency?

Comment (Borda) [reply requirements/image.txt:6]: ref: https://github.com/toshas/torch-fidelity/issues/53
cc: @JustinGoheen

**Review by Borda — CHANGES_REQUESTED** [review]
the dependency issue https://github.com/Lightning-AI/torchmetrics/pull/1939#discussion_r1283550803 would prevent user from even using the remaining metrics based on `torch-fidelity` so lets restore PyPI dependency and set this master for testing?

**Comment by SkafteNicki** [comment]
@Borda, @justusschock please re-review. After a second look at this, we already had a lot of the needed functionality already, so torch-fidelity is no longer needed for the implementation (only testing). A bit of code needed to be copied, but I have linked where needed to original implementation.

**Comment by SkafteNicki** [comment]
> > After a second look at this, we already had a lot of the needed functionality already, so torch-fidelity is no longer needed for the implementation (only testing). A bit of code needed to be copied, but I have linked where needed to original implementation.
> 
> I am fine with the code, my only concern is the dependency, can we somehow use the released package? what we really need form the master branch?

It's no longer a dependency of this metric, only torchvision is needed.

But we need it for testing to compare against. And we specific need master because the addition of PPL in torch-fidelity happened after latest release

**Review by Borda — APPROVED** [review]

**Pull request merged**